hca/submission.py

```diff
@@ -40,19 +61,31 @@ def contains_entity(self, entity_attributes: dict) -> bool:
             return True
         return False
 
-    def add_accessions_to_attributes(self, entity: Entity):
-        for service, mapping_list in self._accession_spec.items():
-            accession = entity.get_accession(service)
+    def add_accessions_to_attributes(self, entity: Entity, archive_type: str, entity_type: str):
+        accession = entity.get_accession(archive_type)
+
+        if accession:
+            accession_mappers_by_archive = self._accession_spec.get(archive_type)
+            accession_mapper = accession_mappers_by_archive.get(entity_type)
+            accession_location = accession_mapper.mapping
+            accession_type = accession_mapper.type
+            if accession_location:
+                self.__set_accession_by_attribute_location(accession, accession_location, accession_type,
```

Comment:
the setting and getting of accession could be placed in the AccessionMapper class. Then you could add unit tests for them as well.

Reply:
@aaclan-ebi Done and pushed